### PR TITLE
Do not simplify before calling kore-rpc

### PR DESCRIPTION
**Pull request by jberthold**
When an execute request ends with an `Aborted`, `Stuck`, or `Branching` reason, it is usually re-executed with legacy `kore-rpc`. As part of that, the simplifier will run on the input.

However, we are _also_ explicitly simplifying the end state of the execute request (and return `Vacuous` if it simplifies to `\bottom`). This additional work is unnecessary as we will anyway get the `\bottom` result back from `kore-rpc` (alas, in a slightly different shape).

This saves one round of fall-back to `kore-rpc` and should speed up long-running proofs that have to fall back often due to side condition checking.

### CODE REVIEW

File: tools/booster/Proxy.hs [hunk 1]
```diff
@@ -235,85 +235,79 @@ respondEither mbStatsVar forceFallback booster kore req = case req of
...
-                    -- simplify Booster's state with Kore's simplifier
-                    Log.logInfoNS "proxy" . Text.pack $ "Simplifying booster state and falling back to Kore "
-                    simplifyResult <- simplifyExecuteState logSettings r._module boosterResult.state
-                    case simplifyResult of
-                        Left logsOnly -> do
-                            -- state was simplified to \bottom, return vacuous
-                            Log.logInfoNS "proxy" "Vacuous state after simplification"
-                            pure . Right . Execute $ makeVacuous logsOnly boosterResult
-                        Right (simplifiedBoosterState, boosterStateSimplificationLogs) -> do
-                            -- attempt to do one step in the old backend
```
Comment (jberthold) [inline tools/booster/Proxy.hs:247]: Essentially, this part of the code has been cut out. The rest is re-indentation.

**Comment by jberthold** [comment]
### Kontrol comparison

| Test                                                                       | HOTFIX-avoid-kore-simplification time | master time       | (HOTFIX-avoid-kore-simplification/master) time 
|----------------------------------------------------------------------------|---------------------------------------|-------------------|------------------------------------------------
| LoopsTest.sum_N(uint256)                                                   | 127.05                                | 163.81            | 0.7755936755997802                             
| AllowChangesTest.testAllow()                                               | 108.04                                | 135.17            | 0.7992897832359253                             
| AllowChangesTest.testFailAllowCallsToAddress()                             | 106.4                                 | 132.17            | 0.8050238329424227                             
| AllowChangesTest.testFailAllowChangesToStorage()                           | 108.42                                | 131.82            | 0.8224852071005918                             
| CounterTest.testSetNumber(uint256)                                         | 124.83                                | 151.29            | 0.8251041046995836                             
| AccountParamsTest.test_getNonce_unknownSymbolic(address)                   | 90.5                                  | 109.31            | 0.827920592809441                              
| PlainPrankTest.test_startPrankWithOrigin_true()                            | 106.96                                | 128.5             | 0.8323735408560311                             
| AccountParamsTest.testNonceSymbolic(uint64)                                | 82.5                                  | 99.11             | 0.832408435072142                              
| PlainPrankTest.test_startPrank_zeroAddress_true()                          | 112.02                                | 133.93            | 0.8364070783244978                             
| ExpectRevertTest.test_ExpectRevert_increasedDepth()                        | 128.17                                | 152.92            | 0.8381506670154328                             
| AssumeTest.test_multi_assume(address,address)                              | 110.41                                | 131.57            | 0.8391730637683362                             
| AssertTest.test_assert_true_branch(uint256)                                | 72.36                                 | 85.87             | 0.8426691510422731                             
| StoreTest.testStoreLoad()                                                  | 70.64                                 | 83.54             | 0.8455829542734019                             
| PlainPrankTest.test_startPrank_true()                                      | 108.43                                | 127.57            | 0.8499647252488831                             
| CounterTest.testIncrement()                                                | 129.37                                | 151.33            | 0.8548866715125883                             
| Setup2Test.testFail_setup()                                                | 77.54                                 | 90.68             | 0.8550948389942655                             
| PlainPrankTest.test_prank_zeroAddress_true()                               | 119.27                                | 138.83            | 0.8591082619030468                             
| ExpectRevertTest.testFail_ExpectRevert_failAndSuccess()                    | 80.49                                 | 93.22             | 0.8634413216048058                             
| EmitContractTest.testExpectEmit()                                          | 72.94                                 | 84.3              | 0.8652431791221827                             
| BytesTypeTest.test_bytes4(bytes4)                                          | 39.66                                 | 45.43             | 0.8729914153642967                             
| ExpectCallTest.testExpectStaticCall()                                      | 75.11                                 | 86.03             | 0.8730675345809601                             
| SetUpDeployTest.test_extcodesize()                                         | 83.64                                 | 95.61             | 0.874803890806401                              
| AddrTest.test_notBuiltinAddress_symbolic(address)                          | 57.97                                 | 66.26             | 0.8748868095381829                             
| SymbolicStorageTest.testFail_SymbolicStorage1(uint256)                     | 208.32                                | 237.32            | 0.8778021237148155                             
| ExpectRevertTest.test_expectRevert_true()                                  | 80.02                                 | 91.12             | 0.8781826163301141                             
| SymbolicStorageTest.testFail_SymbolicStorage(uint256)                      | 171.68                                | 195.05            | 0.8801845680594719                             
| AccountParamsTest.testFail_GetNonce_true()                                 | 61.39                                 | 69.72             | 0.8805220883534137                             
| BMCLoopsTest.test_countdown_symbolic(uint256)                              | 118.68                                | 134.77            | 0.8806114120353195                             
| EmitContractTest.testExpectEmitCheckEmitter()                              | 73.61                                 | 83.43             | 0.8822965360182188                             
| ExpectRevertTest.test_expectRevert_message()                               | 85.59                                 | 96.71             | 0.8850170613173406                             
| ExpectCallTest.testExpectRegularCall()                                     | 75.18                                 | 84.9              | 0.8855123674911661                             
| ArithmeticTest.test_max2(uint256,uint256)                                  | 63.97                                 | 72.23             | 0.8856430845908901                             
| BlockParamsTest.testWarp(uint256)                                          | 42.63                                 | 48.03             | 0.8875702685821362                             
| StoreTest.testStoreLoadNonExistent()                                       | 49.78                                 | 56.03             | 0.8884526146707121                             
| EmitContractTest.testExpectEmitDoNotCheckData()                            | 74.92                                 | 84.3              | 0.8887307236061684                             
| ExpectRevertTest.test_expectRevert_returnValue()                           | 109.8                                 | 123.25            | 0.8908722109533468                             
| ExpectRevertTest.testFail_expectRevert_multipleReverts()                   | 98.82                                 | 110.55            | 0.8938941655359566                             
| ArithmeticTest.test_max1(uint256,uint256)                                  | 69.1                                  | 77.25             | 0.8944983818770226                             
| ExpectRevertTest.testFail_expectRevert_false()                             | 81.8                                  | 91.24             | 0.8965366067514249                             
| InitCodeTest.test_init()                                                   | 133.21                                | 148.28            | 0.8983679525222552                             
| AssumeTest.testFail_assume_true(uint256,uint256)                           | 72.0                                  | 79.95             | 0.900562851782364                              
| AssumeTest.test_assume_false(uint256,uint256)                              | 75.14                                 | 83.26             | 0.9024741772760029                             
| AccountParamsTest.test_Nonce_NonExistentAddress()                          | 52.64                                 | 58.28             | 0.9032258064516129                             
| PlainPrankTest.testFail_startPrank_internalCall()                          | 51.82                                 | 57.21             | 0.9057857018003845                             
| AccountParamsTest.test_Nonce_ExistentAddress()                             | 50.62                                 | 55.57             | 0.9109231599784056                             
| AssumeTest.test_assume_true(uint256,uint256)                               | 46.27                                 | 50.71             | 0.912443305068034                              
| AccountParamsTest.testDealSymbolic(uint256)                                | 50.6                                  | 55.21             | 0.9165006339431263                             
| AccountParamsTest.testDealConcrete()                                       | 45.96                                 | 50.07             | 0.9179149191132415                             
| StoreTest.testGasStoreWarmUp()                                             | 57.39                                 | 62.52             | 0.9179462571976967                             
| BlockParamsTest.testFee(uint256)                                           | 43.2                                  | 46.78             | 0.9234715690466011                             
| StoreTest.testGasLoadColdVM()                                              | 42.0                                  | 45.4              | 0.9251101321585904                             
| AssumeTest.test_assume_staticCall(bool)                                    | 51.46                                 | 55.58             | 0.9258726160489386                             
| src/tests/integration/test_foundry_prove.py::test_foundry_merge_nodes      | 116.82                                | 125.94            | 0.9275845640781324                             
| StoreTest.testGasStoreColdVM()                                             | 43.58                                 | 46.97             | 0.9278262720885672                             
| SymbolicStorageTest.testEmptyInitialStorage(uint256)                       | 45.11                                 | 48.6              | 0.9281893004115226                             
| AssertTest.test_revert_branch(uint256,uint256)                             | 67.51                                 | 72.46             | 0.9316864476952803                             
| AssertTest.test_failing_branch(uint256)                                    | 67.44                                 | 72.24             | 0.9335548172757475                             
| StoreTest.testLoadNonExistent()                                            | 43.07                                 | 46.09             | 0.9344760251681492                             
| AssumeTest.testFail_assume_false(uint256,uint256)                          | 50.02                                 | 53.52             | 0.9346038863976084                             
| src/tests/integration/test_foundry_prove.py::test_foundry_resume_proof     | 96.52                                 | 103.14            | 0.9358153965483808                             
| StoreTest.testGasLoadWarmUp()                                              | 59.23                                 | 63.22             | 0.9368870610566277                             
| InitCodeTest.testFail_init()                                               | 104.9                                 | 111.86            | 0.9377793670659754                             
| BlockParamsTest.testChainId(uint256)                                       | 44.91                                 | 47.38             | 0.9478682988602785                             
| BlockParamsTest.testRoll(uint256)                                          | 44.82                                 | 47.16             | 0.9503816793893131                             
| AccountParamsTest.testEtchConcrete()                                       | 76.65                                 | 80.54             | 0.9517010181276384                             
| src/tests/integration/test_foundry_prove.py::test_foundry_auto_abstraction | 38.2                                  | 40.07             | 0.9533316695782381                             
| LabelTest.testLabel()                                                      | 38.3                                  | 39.99             | 0.9577394348587146                             
| Setup2Test.test_setup()                                                    | 41.74                                 | 43.57             | 0.9579986229056691                             
| PlainPrankTest.test_stopPrank_notExistent()                                | 40.4                                  | 42.15             | 0.9584816132858838                             
| BMCLoopsTest.test_countdown_concrete()                                     | 35.7                                  | 33.97             | 1.0509272887842216                             
| TOTAL                                                                      | 5485.239999999999                     | 6241.859999999999 | 0.8787829268839736

File: scripts/compare.py [hunk 2]
```diff
@@ -1,21 +1,36 @@
...
+    minchange = 0.035
+
+testname = re.compile(r'^src/.*\[(?P<name>.*)\]$')
+
+def readName(input):
+    r = testname.match(input)
+    if r is None:
+        return input
+    else:
+        return r.group("name")
```
Comment (jberthold) [inline scripts/compare.py:24]: I think the names will be unambiguous so we should not print the path to the python runner.

File: test/rpc-integration/test-vacuous/response-vacuous-at-branch.json [hunk 3]
```diff
@@ -9,141 +9,147 @@
...
-                    "name": "Lbl'-LT-'generatedTop'-GT-'",
-                    "sorts": [],
-                    "args": [
+                    "tag": "And",
+                    "sort": {
+                        "tag": "SortApp",
+                        "name": "SortGeneratedTopCell",
+                        "args": []
+                    },
+                    "patterns": [
```
Comment (jberthold) [inline test/rpc-integration/test-vacuous/response-vacuous-at-branch.json:18]: Unfortunately the `kore-rpc` responses for `Vacuous` results just echo the input, not splitting it into term and predicate.

**Review by geo2a — APPROVED** [review]
Great job cleaning this up! Kontrol performance results look good to me.

**Comment by jberthold** [comment]
## KEVM comparison

| Test                                      | HOTFIX-avoid-kore-simplification time | master time       | (HOTFIX-avoid-kore-simplification/master) time 
|-------------------------------------------|---------------------------------------|-------------------|------------------------------------------------
| mcd/end-subuu-pass-spec.k                 | 67.29                                 | 143.81            | 0.4679090466587859                             
| erc20/ds/balanceOf-spec.k                 | 44.61                                 | 62.21             | 0.7170872850024111                             
| mcd/vat-subui-pass-rough-spec.k           | 159.03                                | 218.98            | 0.7262307060005481                             
| mcd/vat-addui-pass-spec.k                 | 159.24                                | 217.99            | 0.730492224413964                              
| mcd/vat-subui-fail-rough-spec.k           | 109.65                                | 148.49            | 0.7384335645498014                             
| mcd/vat-addui-fail-rough-spec.k           | 111.5                                 | 150.3             | 0.7418496340652029                             
| mcd/vat-mului-pass-spec.k                 | 123.46                                | 165.09            | 0.7478345145072384                             
| mcd/vat-muluu-pass-spec.k                 | 126.45                                | 167.32            | 0.7557375089648578                             
| mcd/vat-slip-pass-rough-spec.k            | 202.71                                | 264.78            | 0.7655789712213914                             
| benchmarks/address00-spec.k               | 43.48                                 | 56.07             | 0.7754592473693597                             
| benchmarks/ecrecover00-siginvalid-spec.k  | 78.93                                 | 101.3             | 0.7791707798617967                             
| mcd/vat-subui-pass-spec.k                 | 219.32                                | 279.31            | 0.7852207224947191                             
| mcd/flipper-bids-pass-rough-spec.k        | 132.51                                | 167.61            | 0.7905852872740289                             
| mcd/vat-move-diff-rough-spec.k            | 151.74                                | 184.36            | 0.8230635712735951                             
| bihu/forwardToHotWallet-success-1-spec.k  | 57.8                                  | 68.85             | 0.8395061728395062                             
| mcd/cat-exhaustiveness-spec.k             | 122.12                                | 143.53            | 0.8508325785550059                             
| mcd/dai-adduu-fail-rough-spec.k           | 60.15                                 | 70.59             | 0.8521036974075648                             
| mcd/flipper-addu48u48-fail-rough-spec.k   | 63.43                                 | 73.67             | 0.861001764626035                              
| bihu/forwardToHotWallet-failure-4-spec.k  | 94.46                                 | 108.48            | 0.8707595870206489                             
| mcd/dstoken-approve-fail-rough-spec.k     | 93.6                                  | 106.06            | 0.8825193286818781                             
| examples/solidity-code-spec.md            | 176.78                                | 198.93            | 0.8886543005077162                             
| bihu/forwardToHotWallet-failure-3-spec.k  | 74.87                                 | 83.91             | 0.8922655225837207                             
| bihu/forwardToHotWallet-failure-2-spec.k  | 61.99                                 | 69.29             | 0.8946456920190503                             
| mcd/dsvalue-peek-pass-rough-spec.k        | 59.84                                 | 66.47             | 0.9002557544757034                             
| mcd/dsvalue-read-pass-spec.k              | 58.26                                 | 64.43             | 0.9042371566040663                             
| erc20/hkg/transferFrom-failure-1-spec.k   | 75.82                                 | 82.87             | 0.9149269940871243                             
| erc20/ds/transfer-failure-1-b-spec.k      | 87.3                                  | 95.18             | 0.917209497793654                              
| bihu/forwardToHotWallet-success-2-spec.k  | 122.11                                | 133.12            | 0.9172926682692307                             
| mcd/flipper-ttl-pass-spec.k               | 57.37                                 | 62.49             | 0.9180668907025124                             
| erc20/ds/transfer-success-1-spec.k        | 67.2                                  | 73.01             | 0.9204218600191755                             
| erc20/ds/transferFrom-failure-2-a-spec.k  | 71.16                                 | 77.27             | 0.9209266209395626                             
| erc20/hkg/transferFrom-failure-2-spec.k   | 82.9                                  | 89.97             | 0.9214182505279538                             
| mcd/flopper-cage-pass-spec.k              | 59.47                                 | 64.38             | 0.9237340789064927                             
| erc20/ds/transferFrom-success-2-spec.k    | 80.72                                 | 87.33             | 0.9243100881713042                             
| erc20/ds/transferFrom-failure-1-b-spec.k  | 128.56                                | 138.9             | 0.9255579553635709                             
| mcd/vat-deny-diff-fail-rough-spec.k       | 88.36                                 | 95.24             | 0.9277614447711047                             
| erc20/hkg/transfer-success-2-spec.k       | 51.44                                 | 55.38             | 0.928855182376309                              
| erc20/hkg/transferFrom-success-1-spec.k   | 68.42                                 | 73.52             | 0.9306311207834603                             
| erc20/ds/transferFrom-failure-1-c-spec.k  | 98.08                                 | 105.29            | 0.931522461772248                              
| erc20/ds/transfer-failure-1-a-spec.k      | 67.53                                 | 72.47             | 0.9318338622878433                             
| erc20/ds/transferFrom-failure-1-a-spec.k  | 82.39                                 | 88.32             | 0.9328577898550725                             
| mcd/flipper-tau-pass-spec.k               | 58.07                                 | 62.22             | 0.9333011893281903                             
| erc20/ds/transfer-success-2-spec.k        | 63.92                                 | 68.44             | 0.9339567504383403                             
| erc20/ds/transferFrom-failure-2-b-spec.k  | 86.78                                 | 92.63             | 0.9368455144121776                             
| erc20/ds/transferFrom-success-1-spec.k    | 92.93                                 | 99.15             | 0.9372667675239537                             
| benchmarks/storagevar02-overflow-spec.k   | 43.28                                 | 46.03             | 0.9402563545513796                             
| erc20/hkg/transfer-failure-2-spec.k       | 53.2                                  | 56.47             | 0.9420931468036126                             
| bihu/forwardToHotWallet-failure-1-spec.k  | 45.09                                 | 47.69             | 0.9454812329628854                             
| erc20/hkg/transfer-success-1-spec.k       | 54.42                                 | 57.35             | 0.9489102005231037                             
| mcd/cat-file-addr-pass-rough-spec.k       | 57.77                                 | 60.78             | 0.9504771306350773                             
| benchmarks/storagevar02-nooverflow-spec.k | 44.52                                 | 46.69             | 0.9535232383808097                             
| examples/storage-spec.md                  | 163.7                                 | 171.65            | 0.9536848237692979                             
| functional/evm-int-simplifications-spec.k | 97.02                                 | 101.7             | 0.9539823008849557                             
| benchmarks/encode-keccak00-spec.k         | 49.9                                  | 52.23             | 0.9553896228221329                             
| mcd/dai-symbol-pass-spec.k                | 50.14                                 | 52.38             | 0.9572355861015654                             
| erc20/ds/transfer-failure-2-a-spec.k      | 66.52                                 | 69.27             | 0.960300274289014                              
| erc20/hkg/allowance-spec.k                | 42.98                                 | 44.73             | 0.9608763693270735                             
| mcd/functional-spec.k                     | 417.99                                | 434.77            | 0.9614048807415415                             
| erc20/ds/approve-failure-spec.k           | 46.28                                 | 48.01             | 0.9639658404499063                             
| erc20/ds/transferFrom-failure-2-c-spec.k  | 53.67                                 | 55.61             | 0.9651141880956663                             
| benchmarks/storagevar01-spec.k            | 59.28                                 | 61.38             | 0.9657869012707723                             
| erc20/hkg/transfer-failure-1-spec.k       | 55.46                                 | 57.41             | 0.9660337920222959                             
| examples/erc20-spec.md                    | 206.9                                 | 193.76            | 1.0678158546655658                             
| benchmarks/ecrecover00-sigvalid-spec.k    | 101.56                                | 95.11             | 1.0678162128062243                             
| erc20/ds/transfer-failure-2-b-spec.k      | 62.64                                 | 49.64             | 1.2618855761482675                             
| benchmarks/bytes00-spec.k                 | 69.43                                 | 54.77             | 1.2676647799890453                             
| mcd/dsvalue-read-pass-summarize-spec.k    | 111.72                                | 59.09             | 1.8906752411575563                             
| TOTAL                                     | 6197.220000000002                     | 7015.529999999997 | 0.8833573514759405

**Pull request merged**